components/brave_wallet/browser/solana_compiled_instruction.h

```diff
@@ -0,0 +1,74 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_BRAVE_WALLET_BROWSER_SOLANA_COMPILED_INSTRUCTION_H_
+#define BRAVE_COMPONENTS_BRAVE_WALLET_BROWSER_SOLANA_COMPILED_INSTRUCTION_H_
+
+#include <string>
+#include <vector>
+
+#include "third_party/abseil-cpp/absl/types/optional.h"
+
+namespace brave_wallet {
+
+class SolanaAddress;
+class SolanaInstruction;
+class SolanaMessageAddressTableLookup;
+
+// https://docs.rs/solana-sdk/1.14.12/solana_sdk/instruction/struct.CompiledInstruction.html
+class SolanaCompiledInstruction {
+ public:
+  SolanaCompiledInstruction(uint8_t program_id_index,
+                            const std::vector<uint8_t>& account_indexes,
+                            const std::vector<uint8_t>& data);
+  SolanaCompiledInstruction(const SolanaCompiledInstruction&) = delete;
+  SolanaCompiledInstruction(SolanaCompiledInstruction&&);
+  SolanaCompiledInstruction& operator=(const SolanaCompiledInstruction&) =
+      delete;
+  SolanaCompiledInstruction& operator=(SolanaCompiledInstruction&&);
+  ~SolanaCompiledInstruction();
+
+  bool operator==(const SolanaCompiledInstruction& ins) const;
+
+  void Serialize(std::vector<uint8_t>* bytes) const;
+  static absl::optional<SolanaCompiledInstruction> Deserialize(
+      const std::vector<uint8_t>& bytes,
+      size_t* bytes_index);
+
+  static absl::optional<SolanaCompiledInstruction> FromInstruction(
+      const SolanaInstruction& instruction,
+      const std::vector<SolanaAddress>& static_accounts,
+      const std::vector<SolanaMessageAddressTableLookup>& addr_table_lookups,
+      uint8_t num_of_write_indexes,
+      uint8_t num_of_read_indexes);
```

Comment:
It would be clearer if these are renamed to `num_of_total_write_indexes` and `num_of_total_read_indexes`, otherwise we have to check example caller to see that is actually sum of indexes in lookups.

Reply:
768cd4e79422010dfee3015d7d594b1701521f62
Also found num_read_indexes is unused now so removed.